### ISSUE TITLE
[RyuJIT/arm32][GCStress=4] Failures in GCStress run

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#14383]
I ran the RyuJIT/arm32 pri-1 tests on Windows hardware, with `COMPlus_GCStress=4`, and saw 204 failures, 10942 pass.

Many (but not all) of the failures are the common assert indicating bad GC information:
```
TEST_CMD_LINE: JIT\jit64\gc\misc\structret6_3> C:\brucefo\Windows_NT.arm.Release.2017-10-06\JIT\jit64\gc\misc\structret6_3\structret6_3.cmd 

BEGIN EXECUTION
 "c:\brucefo\core_root\Windows_NT.arm.Checked.2017-10-06\corerun.exe" structret6_3.exe 

Assert failure(PID 7592 [0x00001da8], Thread: 6776 [0x1a78]): !CREATE_CHECK_STRING(!"Detected use of a corrupted OBJECTREF. Possible GC hole.")

!  + 0x0 (0x00000000)
    File: e:\gh\coreclr2\src\vm\object.cpp Line: 1819
    Image: c:\brucefo\core_root\Windows_NT.arm.Checked.2017-10-06\CoreRun.exe

Expected: 100
Actual: 123456789
END EXECUTION - FAILED
FAILED
```

Many of the tests look to be related to structs, or exception handling.

There are likely to be multiple different causes for these failures.

All the failed tests:
```
Tests.lst=_dbgthrow.cmd_7830
Tests.lst=_il_dbgcastclass_catch.cmd_6797
Tests.lst=_il_dbgfilter.cmd_11282
Tests.lst=_il_dbghuge_gcref.cmd_98
Tests.lst=_il_dbghuge_struct.cmd_1734
Tests.lst=_il_dbgisinst_catch.cmd_9146
Tests.lst=_il_dbgtest_implicit.cmd_10113
Tests.lst=_il_relcastclass_catch.cmd_4232
Tests.lst=_il_relfilter.cmd_2245
Tests.lst=_il_relisinst_catch.cmd_11096
Tests.lst=_il_relknight.cmd_4493
Tests.lst=_il_relrefarg_i1.cmd_10887
Tests.lst=_il_relrefloc_i2.cmd_6208
Tests.lst=_il_relrefloc_o.cmd_40
Tests.lst=_il_relrefloc_o2.cmd_5665
Tests.lst=_il_relrefloc_r4.cmd_8718
Tests.lst=_il_relrefloc_r8.cmd_2528
Tests.lst=_il_relrefloc_u2.cmd_8947
Tests.lst=_il_reltest_implicit.cmd_806
Tests.lst=_il_relthread-race.cmd_6666
Tests.lst=_relthrow.cmd_10975
Tests.lst=_speed_dbgthrow.cmd_10067
Tests.lst=_speed_relthrow.cmd_631
Tests.lst=437044.cmd_10304
Tests.lst=ArrayCreateInstance2b.cmd_3144
Tests.lst=b02345.cmd_4358
Tests.lst=b07900.cmd_3881
Tests.lst=b112982.cmd_8489
Tests.lst=b13569.cmd_10869
Tests.lst=b16399.cmd_4334
Tests.lst=b26496.cmd_9315
Tests.lst=b302558.cmd_3591
Tests.lst=b309539.cmd_6644
Tests.lst=b309576.cmd_37
Tests.lst=b321799.cmd_1634
Tests.lst=b33183.cmd_4258
Tests.lst=b441487.cmd_10976
Tests.lst=Baadbaad.cmd_2370
Tests.lst=BaseClass.cmd_6478
Tests.lst=binarytrees.cmd_7301
Tests.lst=binarytrees3.cmd_4184
Tests.lst=bindhandleinvalid.cmd_7988
Tests.lst=bindhandleinvalid3.cmd_11143
Tests.lst=bindhandleinvalid4.cmd_2092
Tests.lst=bindhandleinvalid5.cmd_318
Tests.lst=bindhandleinvalid6.cmd_2744
Tests.lst=boxunboxvaluetype_do.cmd_7
Tests.lst=boxunboxvaluetype_r.cmd_6226
Tests.lst=boxunboxvaluetype_ro.cmd_6993
Tests.lst=calli_excep.cmd_1064
Tests.lst=caninline_d.cmd_6344
Tests.lst=caninline_do.cmd_3927
Tests.lst=caninline_r.cmd_8672
Tests.lst=caninline_ro.cmd_9049
Tests.lst=Case1.cmd_8560
Tests.lst=case1.cmd_9154
Tests.lst=case11.cmd_8858
Tests.lst=case12.cmd_5538
Tests.lst=case14.cmd_262
Tests.lst=case15.cmd_8622
Tests.lst=case2.cmd_10566
Tests.lst=case3.cmd_10915
Tests.lst=case4.cmd_4484
Tests.lst=case5.cmd_6196
Tests.lst=case6.cmd_10325
Tests.lst=case7.cmd_5508
Tests.lst=case8.cmd_2804
Tests.lst=case9.cmd_10923
Tests.lst=castclassvaluetype_d.cmd_2095
Tests.lst=castclassvaluetype_do.cmd_6021
Tests.lst=castclassvaluetype_r.cmd_9730
Tests.lst=castclassvaluetype_ro.cmd_8242
Tests.lst=CircularCctorFourThreads.cmd_5035
Tests.lst=CircularCctorThreeThreads03.cmd_3051
Tests.lst=client.cmd_986
Tests.lst=CollidedUnwind.cmd_8131
Tests.lst=ddb87766.cmd_5825
Tests.lst=Dev11_4421.cmd_10658
Tests.lst=DictEnumIDictEnumget_Entry.cmd_3872
Tests.lst=DictEnumIDictEnumget_Key.cmd_8982
Tests.lst=DictEnumIDictEnumget_Value.cmd_5520
Tests.lst=DictEnumIEnumget_Current.cmd_11133
Tests.lst=DictionaryICollectionCopyTo.cmd_1565
Tests.lst=DictionaryICollectionCopyTo2.cmd_8759
Tests.lst=EHPatternTests.cmd_11092
Tests.lst=excepobj.cmd_7191
Tests.lst=exceptions.cmd_5701
Tests.lst=fannkuch-redux.cmd_11386
Tests.lst=fasta.cmd_11039
Tests.lst=fastaredux.cmd_10129
Tests.lst=Finalizer.cmd_6081
Tests.lst=GCKeepAlive.cmd_8869
Tests.lst=general_class_instance01.cmd_5494
Tests.lst=general_class_static01.cmd_8510
Tests.lst=general_struct_instance01.cmd_7469
Tests.lst=general_struct_static01.cmd_577
Tests.lst=GetGeneration_fail.cmd_10025
Tests.lst=GoryManagedPresent.cmd_5766
Tests.lst=GoryNativePast.cmd_3924
Tests.lst=HandlerException.cmd_3191
Tests.lst=hfa_sd0A_r.cmd_7167
Tests.lst=hfa_sd0B_d.cmd_7353
Tests.lst=hfa_sd0B_r.cmd_4438
Tests.lst=hfa_sd0G_r.cmd_10723
Tests.lst=hfa_sd1A_r.cmd_525
Tests.lst=hfa_sd1G_r.cmd_1954
Tests.lst=hfa_sd2A_d.cmd_8260
Tests.lst=hfa_sd2A_r.cmd_8128
Tests.lst=hfa_sd2B_d.cmd_5688
Tests.lst=hfa_sd2B_r.cmd_752
Tests.lst=hfa_sd2G_d.cmd_4210
Tests.lst=hfa_sd2G_r.cmd_1362
Tests.lst=hfa_sf0A_r.cmd_10307
Tests.lst=hfa_sf0B_d.cmd_5236
Tests.lst=hfa_sf0B_r.cmd_2860
Tests.lst=hfa_sf1A_r.cmd_7852
Tests.lst=hfa_sf2A_d.cmd_5804
Tests.lst=hfa_sf2A_r.cmd_7031
Tests.lst=hfa_sf2B_d.cmd_3846
Tests.lst=hfa_sf2B_r.cmd_7325
Tests.lst=huge_struct.cmd_8601
Tests.lst=InlineThrow.cmd_2281
Tests.lst=jaggedarr_cs_do.cmd_3344
Tests.lst=LdfldaHack.cmd_10205
Tests.lst=lifetime1.cmd_3806
Tests.lst=lifetime2.cmd_9676
Tests.lst=mandelbrot.cmd_11382
Tests.lst=MarshalClassTests.cmd_5485
Tests.lst=moduleHandleCache.cmd_2246
Tests.lst=MultipleException.cmd_5527
Tests.lst=MutualThdRecur-fptr.cmd_5907
Tests.lst=nbody.cmd_11148
Tests.lst=NestedEx1.cmd_3353
Tests.lst=NestedEx2.cmd_4313
Tests.lst=NestedException.cmd_1353
Tests.lst=NoGC.cmd_11390
Tests.lst=OOMException01.cmd_4785
Tests.lst=override_override1.cmd_8398
Tests.lst=Pending.cmd_5724
Tests.lst=pi-digits.cmd_4164
Tests.lst=pumpFromCctor.cmd_1154
Tests.lst=Recurse.cmd_5745
Tests.lst=RecursiveException.cmd_7608
Tests.lst=RecursiveRethrow.cmd_6185
Tests.lst=RecursiveThrowNew.cmd_1299
Tests.lst=regexdna.cmd_5103
Tests.lst=repro177066.cmd_2212
Tests.lst=RethrowAndFinally.cmd_9249
Tests.lst=revcomp.cmd_3664
Tests.lst=self_override3.cmd_11010
Tests.lst=semdemo.cmd_1794
Tests.lst=semtest.cmd_5191
Tests.lst=spectralnorm.cmd_7789
Tests.lst=StackTracePreserveTests.cmd_901
Tests.lst=straccess2_cs_do.cmd_3513
Tests.lst=straccess2_cs_ro.cmd_2232
Tests.lst=stringarr_cs_ro.cmd_6254
Tests.lst=StringCompareOrdinal2.cmd_1273
Tests.lst=struct9_2.cmd_5807
Tests.lst=structret1_1.cmd_5198
Tests.lst=structret1_2.cmd_7200
Tests.lst=structret1_3.cmd_5062
Tests.lst=structret2_1.cmd_4001
Tests.lst=structret2_2.cmd_1242
Tests.lst=structret2_3.cmd_3507
Tests.lst=structret3_1.cmd_3198
Tests.lst=structret3_2.cmd_4020
Tests.lst=structret3_3.cmd_8029
Tests.lst=structret4_1.cmd_10882
Tests.lst=structret4_2.cmd_11286
Tests.lst=structret4_3.cmd_3274
Tests.lst=structret5_1.cmd_10355
Tests.lst=structret5_2.cmd_4782
Tests.lst=structret5_3.cmd_809
Tests.lst=structret6_1.cmd_2824
Tests.lst=structret6_2.cmd_4235
Tests.lst=structret6_3.cmd_420
Tests.lst=switch6.cmd_4866
Tests.lst=test1.cmd_9881
Tests.lst=test532654_b.cmd_1907
Tests.lst=test640474.cmd_6259
Tests.lst=TestAPIs.cmd_939
Tests.lst=threadstatic07.cmd_7048
Tests.lst=throw_cs_d.cmd_10584
Tests.lst=throw_cs_d.cmd_3208
Tests.lst=throw_cs_do.cmd_2138
Tests.lst=throw_cs_do.cmd_2821
Tests.lst=throw_cs_r.cmd_10973
Tests.lst=throw_cs_r.cmd_2662
Tests.lst=throw_cs_ro.cmd_10070
Tests.lst=throw_cs_ro.cmd_6786
Tests.lst=ThrowInCatch.cmd_8628
Tests.lst=ThrowInFinally.cmd_5152
Tests.lst=ThrowInFinallyNestedInTry.cmd_9693
Tests.lst=TryCatchInFinally.cmd_608
Tests.lst=trythrowexcept_d.cmd_3661
Tests.lst=trythrowexcept_r.cmd_484
Tests.lst=UserExceptionThread.cmd_650
Tests.lst=Vector3Interop_r.cmd_7398
Tests.lst=Vector3Interop_ro.cmd_2124
Tests.lst=vsw529206ModuleCctor.cmd_5359
Tests.lst=vsw529206ModuleCctor.cmd_5726
Tests.lst=vsw529206StaticCctor.cmd_7060
Tests.lst=waitallex1.cmd_3595
```

**Comment by BruceForstall** [comment]
@dotnet/arm32-contrib

**Comment by rartemev** [comment]
Required env
set COMPlus_GCStress=4
set COMPlus_AltJit=*
set COMPlus_AltJitNgen=*
set COMPlus_AltJitName=protojit.dll
set COMPlus_NoGuiOnAssert=1
set COMPlus_AltJitAssertOnNYI=1
set COMPlus_AltJitSkipOnAssert=0
set COMPlus_ContinueOnAssert=0

**Comment by rartemev** [comment]
It seems like there are other people how desire so much to work on this issue so I am not going to interfere with them and remove my assignment.

**Comment by BruceForstall** [comment]
Of the tests above, after #14600 there are 122 failures in GCStress=4. However, almost all the ones I looked at (maybe half) have `No output was generated!`, and no obvious GCStress assert.

**Comment by BruceForstall** [comment]
Closing this in favor of ##14856
